### PR TITLE
fix(daemon): ship baseline OpenCode for Windows and stop retrying AVX2 illegal-instruction crashes

### PR DESCRIPTION
**Pull request by lefarcen**
Part of #3408.

## Why

PostHog shows 126 `fatal_rpc_error` events accounting for 41.6% of real AMR failures; 105 of the 129 matching Langfuse traces are Windows binary crashes before OpenCode readiness, and 88 of those (58 Bun `Illegal instruction` traces whose crash payload reports `no_avx2`, plus 30 raw `0xc000001d` exits — the same NTSTATUS without Bun's crash handler catching it) are one root cause: the `opencode.exe` bundled via `@powerformer/vela-cli` was a Bun build that **requires AVX2**. CPUs without AVX2 (Intel Atom/Celeron/Pentium N-series sold through 2021, pre-Haswell desktops — common on low-end Windows laptops) crash on launch, every time.

The daemon made it worse: these crashes arrive with a `runtime_close: fatal_rpc_error` diagnostic, and that close-reason promotion ran BEFORE the existing crash-text detection (`isProcessCrashText` → `process_crashed`), so a deterministic hardware-incompatibility crash was classified as a retryable `fatal_rpc_error`. The automatic retry re-ran the same binary on the same CPU — 105 of the 126 events retried once and failed again, doubling the failed-run count and delaying the (useless) error.

The fix has two halves; after rebasing on main this PR carries only the second:

1. **Root fix (already on main, not in this PR's diff)** — the baseline (non-AVX2) OpenCode bundle ships through `@powerformer/vela-cli`. This branch originally bumped it to `0.0.25` (the first stable cut after powerformer/vela#838); main has since moved to `0.0.26` (also a baseline build — verified in its release CI log: `--target=windows-x64-baseline`), so the conflict resolution dropped this PR's version bump as superseded. The vela-side hardening from #838 remains in force: the post-publish `smoke-win32` job boots the published bundle on a real Windows runner, and the lockfile regenerator pins the baseline asset with a unit test so a routine version bump cannot silently flip back (that flip is exactly what happened to `0.0.24`, which was cut before #838 merged).
2. **Classification fix (this PR)** — the crash text now classifies as `process_exit`/`cpu_unsupported` ahead of the `fatal_rpc_error` close-reason promotion, is never auto-retried, and gets a dedicated error card. Matching is deliberately narrow: the `no_avx2` CPU-feature line from Bun's crash banner matches unconditionally (the feature line is itself the proof), while the raw Windows STATUS_ILLEGAL_INSTRUCTION forms (hex `0xC000001D` / Go's decimal rendering `3221225501`) only match inside vela's bundled-opencode startup wrapper text (`start opencode server` / `opencode exited before readiness`) — a generic SIGILL from any other agent binary stays on the retryable path. This keeps users on older app versions and any future runtime binary with the same problem diagnosable in PostHog (`cpu_unsupported` vs the opaque retryable bucket) and stops the retry doubling.

## What users will see

- Windows users on CPUs without AVX2 (previously: every AMR run failed twice with an opaque "error" card and a useless Retry) can now run AMR normally — the bundled runtime no longer requires AVX2.
- If an agent binary ever crashes with an illegal instruction again (e.g. an old app version), the chat error card now says "Processor not supported" with copy explaining the CPU lacks an instruction set the runtime needs and pointing at updating Open Design, instead of a dead Retry button. No automatic retry is attempted.

## Surface area

- [x] **UI** — new error-card variant (title + message) in the chat failure card, `apps/web`
- [ ] **Keyboard shortcut** — new or changed
- [ ] **CLI / env var**
- [x] **API / contract** — new `TrackingRunFailureDetail` value `cpu_unsupported` in `packages/contracts` (additive enum member; no endpoint/SSE shape change)
- [ ] **Extension point**
- [x] **i18n keys** — `chat.runError.title.cpuUnsupported` + `chat.runError.cpuUnsupportedMessage`, translated in all 19 locales
- [ ] **New top-level dependency** — (no dependency changes; the `@powerformer/vela-cli` bump this branch once carried was superseded by main's `0.0.26`)
- [x] **Default behavior change** — unsupported-CPU crashes are no longer auto-retried and show a dedicated error card (the baseline OpenCode bundle itself ships via main's `vela-cli 0.0.26`)
- [ ] **None**

## Screenshots

Captured from a real end-to-end reproduction (real daemon on this branch + real vela signed into prod + `VELA_OPENCODE_BIN` pointed at a stub that replays the Windows crash shape — Bun crash banner with `no_avx2` on stderr, immediate death before readiness). Sent from the chat composer like a normal user turn; vela reported `start opencode server: opencode exited before readiness`, the daemon classified `failureDetail: cpu_unsupported`, retry was suppressed (`retry_suppressed_reason: non_retryable_category` in the run events), and the chat rendered the new card — "处理器不支持" title, update guidance, no Retry button, raw error preserved in the collapsible source row. Screenshot attached in the PR conversation (uploaded manually — CLI can't attach images).

**Instruction-set level A/B/C proof** (QEMU user-mode TCG, `-cpu Nehalem` = SSE4.2 without AVX/AVX2 — the exact CPU family Bun's baseline target serves): the fork's NON-baseline build dies with `Illegal instruction` (exit 132, reproducing the production crash) on the emulated no-AVX2 CPU; the baseline build of the same pinned version prints its version and exits 0 on that same CPU; and the same non-baseline binary runs fine under `-cpu max` (AVX2 present), proving the crash is purely the instruction set. Combined with the Windows-runner boot smoke in powerformer/vela#838, this closes the hardware-compatibility verification without physical no-AVX2 hardware.

A physical no-AVX2 hardware pass would still be the gold standard if a machine turns up (or a Hyper-V VM with Processor Compatibility Mode, which masks AVX): install the 0.15.1 build, run any AMR turn. The card reuses the existing run-failure card component unchanged — only the mapping is new (`cpu_unsupported` → title "Processor not supported", message "This agent's runtime needs a CPU instruction set (AVX2) that this device doesn't have, so it can't start. Update Open Design to the latest version, which ships a compatible runtime.", no Retry button, no AMR switch card). The mapping is covered by `apps/web/tests/runtime/amr-guidance.test.ts`.

## Bug fix verification

- Test paths:
  - `apps/daemon/tests/run-failure-classification.test.ts` — "cpu_unsupported (AVX2) crash classification": Bun illegal-instruction banner under an ACP fatal close, decimal `3221225501` and hex `0xC000001D` inside the opencode startup wrapper, the production abort-after-panic shape (exit status 3), plus negatives: a banner without `no_avx2`, a `3221225501` exit outside the opencode startup context, and plain ACP fatals without crash text all stay `fatal_rpc_error`/retryable.
  - `apps/daemon/tests/run-retry-policy.test.ts` — `cpu_unsupported` is suppressed as `non_retryable_category` even with a leaked `retryable: true` hint.
  - `apps/web/tests/runtime/amr-guidance.test.ts` — `cpu_unsupported` maps to the update-guidance card with no Retry and no switch card, for every agent.
- Red on `main`, green on this branch: **yes** (the three new classification cases returned `fatal_rpc_error`/`retryable: true` on main's behavior).
- The binary-level fix is verified in vela CI: the win32 build log shows `bun run script/build.ts "--target=windows-x64-baseline"`, and the new `smoke-win32` job boots the published bundle's `opencode.exe --version` on a `windows-latest` runner (see powerformer/vela#838).

## Adjacent issues (not in this PR)

- 5 Langfuse traces are `0xc0000139` (STATUS_ENTRYPOINT_NOT_FOUND) — missing DLL entry points on old/unpatched Windows builds. Different root cause (minimum OS baseline, not CPU), needs its own spec.
- 12 traces (7× `0xc0000409` stack-buffer-overrun, 5× Bun internal assertion) keep the existing `process_crashed`/`fatal_rpc_error` handling; not yet proven deterministic.
- `run_finished` telemetry does not carry a CPU-model property, so quantifying the no-AVX2 install base from PostHog still requires Langfuse crash payloads.

## Validation

- `pnpm guard` — pass
- `pnpm typecheck` — pass (all workspaces)
- `pnpm --filter @open-design/daemon exec vitest run tests/run-failure-classification.test.ts tests/run-retry-policy.test.ts tests/run-failure-detail-status.test.ts tests/run-failure-detail-persisted-message.test.ts tests/run-resume-on-failure.test.ts` — 100 passed
- `pnpm --filter @open-design/web exec vitest run tests/runtime/amr-guidance.test.ts tests/components/AmrGuidance.test.tsx` — 23 passed
- `pnpm install` after merging main (which carries the `vela-cli 0.0.26` lockfile) — no drift

### CODE REVIEW

**Comment by lefarcen** [comment]
🧪 This PR changes the user-facing run-failure path, so it will need a manual QA pass before merge. Please hold off on self-merging for now; once the review gates are clear, we’ll queue QA on it.

File: apps/daemon/src/run-failure-classification.ts [hunk 1]
```diff
@@ -440,6 +440,18 @@ function isProcessCrashText(text: string): boolean {
...
+// The child binary executed an instruction this CPU does not implement — in
+// practice a Bun-compiled agent (bundled opencode) built for AVX2 running on a
+// CPU without it (Intel Atom/Celeron/Pentium N-series through 2021). Seen as
+// the Bun crash banner with a `no_avx2` CPU feature line, the POSIX "Illegal
+// instruction" message, or Windows STATUS_ILLEGAL_INSTRUCTION as either the
+// hex 0xC000001D or Go/Node's decimal exit-status rendering 3221225501. The
+// same binary on the same CPU fails deterministically, so this must never be
+// auto-retried.
+function isCpuUnsupportedCrashText(text: string): boolean {
+  return /\billegal instruction\b|\bno_avx2\b|0xc000001d|\b3221225501\b/i.test(text);
```
Comment (nettee) [inline apps/daemon/src/run-failure-classification.ts]: `collectFailureText()` feeds this helper raw stderr and error text from the failed run, so matching any plain `Illegal instruction` line here is much broader than the AVX2 readiness crash described in the PR. A Bun/runtime bug on an AVX2-capable machine, or any other SIGILL surfaced in agent stderr, will now be reclassified from `process_crashed`/`fatal_rpc_error` to `cpu_unsupported`, which suppresses retry and shows the misleading "Processor not supported" guidance. The changed regex is the only evidence needed: `\billegal instruction\b` is sufficient on its own, without `no_avx2`, the Bun CPU-feature banner, or the specific Windows exit status. Please tighten this matcher to the signatures that actually prove the AVX2 case, for example requiring `no_avx2`, the Bun banner plus that CPU-feature line, or the explicit `0xC000001D`/`3221225501` status forms, and add a negative test for a generic `Illegal instruction` crash that should stay `process_crashed`.

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

**Review by nettee — CHANGES_REQUESTED** [review]
I found one blocking issue in the new cpu-unsupported classifier. The packaging bump looks fine, but this matcher needs to stay tightly scoped to the AVX2/no-AVX2 startup crash shape before it suppresses retries and shows the new unsupported-processor card.

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

File: apps/daemon/src/run-failure-classification.ts (hunk 1 above)
Comment (nettee) [inline apps/daemon/src/run-failure-classification.ts]: `collectFailureText()` feeds this matcher raw stderr and error text, so treating any bare `Illegal instruction` line as `cpu_unsupported` is still too broad for the AVX2-only case this PR is trying to carve out. A Bun/runtime bug on an AVX2-capable machine, or any unrelated SIGILL surfaced in agent stderr, will now skip retry and show the misleading "Processor not supported" card even though the CPU may be fine. The changed test at `apps/daemon/tests/run-failure-classification.test.ts` now locks that behavior in by expecting a generic Bun `Illegal instruction` banner with no `no_avx2` line or Windows status code to classify as `cpu_unsupported`. Please tighten this helper to signatures that actually prove the unsupported-CPU shape, for example `no_avx2`, the Bun crash banner plus that CPU-feature line, or the explicit `0xC000001D` / `3221225501` status forms, and add a negative test that a generic `Illegal instruction` crash still stays `process_crashed`.

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

**Review by nettee — CHANGES_REQUESTED** [review]
I found one blocking issue in the current `cpu_unsupported` classifier. The packaging bump looks right, but the daemon still broadens this new failure detail beyond the AVX2 startup crash shape described in the PR.

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

File: apps/daemon/src/run-failure-classification.ts [hunk 2]
```diff
@@ -461,6 +461,18 @@ function isProcessCrashText(text: string): boolean {
...
```
Comment (nettee) [inline apps/daemon/src/run-failure-classification.ts]: `collectFailureText()` feeds this helper raw stderr and error text from the failed run, so `\billegal instruction\b` on its own is still too broad for the AVX2-only crash shape this PR is trying to carve out. With the current regex, any unrelated SIGILL or Bun crash that mentions "Illegal instruction" will now be reclassified from `process_crashed`/`fatal_rpc_error` to `cpu_unsupported`, which suppresses retry and shows the misleading processor-support card even on AVX2-capable machines. The changed lines are the evidence: this helper returns true without `no_avx2`, without the Bun CPU-feature banner, and without the explicit Windows `0xC000001D` / `3221225501` status forms the PR body calls out. Please tighten the matcher to signatures that actually prove the unsupported-CPU case, for example requiring `no_avx2`, the Bun crash banner plus that CPU-feature line, or the explicit Windows status codes, and add a negative test that a generic `Illegal instruction` crash stays `process_crashed`.

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

**Review by nettee — CHANGES_REQUESTED** [review]
The runtime/UI pieces look consistent, but one blocker remains in the classifier: the new `cpu_unsupported` path is still broader than the AVX2/no-AVX2 startup crash shape described in the PR, so it can suppress retries and show the wrong guidance for unrelated SIGILL/Bun crashes.

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

File: apps/daemon/src/run-failure-classification.ts (hunk 1 above)
Comment (lefarcen) [reply apps/daemon/src/run-failure-classification.ts]: Valid — fixed in 525ecad25. `isCpuUnsupportedCrashText` no longer matches a bare `Illegal instruction`; it now requires a precise signal: the `no_avx2` CPU-feature line (which Bun's crash banner always prints on machines actually missing AVX2 — all 58 banner-bearing production traces carry it, including today's exit-status-3 case) or the bare Windows STATUS_ILLEGAL_INSTRUCTION renderings `0xC000001D`/`3221225501` (those runs have no banner at all). A generic illegal-instruction banner without `no_avx2` returns to `process_crashed`/`fatal_rpc_error` with retry intact — the previously-changed test is reverted to expect `process_crashed`, and a new negative spec (banner with `CPU: sse42 avx avx2`) locks the boundary.

Comment (lefarcen) [reply apps/daemon/src/run-failure-classification.ts]: Valid — fixed in 525ecad25. `isCpuUnsupportedCrashText` no longer matches a bare `Illegal instruction`; it now requires a precise signal: the `no_avx2` CPU-feature line (which Bun's crash banner always prints on machines actually missing AVX2 — all 58 banner-bearing production traces carry it, including today's exit-status-3 case) or the bare Windows STATUS_ILLEGAL_INSTRUCTION renderings `0xC000001D`/`3221225501` (those runs have no banner at all). A generic illegal-instruction banner without `no_avx2` returns to `process_crashed`/`fatal_rpc_error` with retry intact — the previously-changed test is reverted to expect `process_crashed`, and a new negative spec (banner with `CPU: sse42 avx avx2`) locks the boundary.

File: apps/daemon/src/run-failure-classification.ts (hunk 2 above)
Comment (lefarcen) [reply apps/daemon/src/run-failure-classification.ts]: Valid — fixed in 525ecad25. `isCpuUnsupportedCrashText` no longer matches a bare `Illegal instruction`; it now requires a precise signal: the `no_avx2` CPU-feature line (which Bun's crash banner always prints on machines actually missing AVX2 — all 58 banner-bearing production traces carry it, including today's exit-status-3 case) or the bare Windows STATUS_ILLEGAL_INSTRUCTION renderings `0xC000001D`/`3221225501` (those runs have no banner at all). A generic illegal-instruction banner without `no_avx2` returns to `process_crashed`/`fatal_rpc_error` with retry intact — the previously-changed test is reverted to expect `process_crashed`, and a new negative spec (banner with `CPU: sse42 avx avx2`) locks the boundary.

**Review by nettee — APPROVED** [review]
@lefarcen I re-reviewed the current head and verified the classifier is now narrowly scoped to the AVX2/no-AVX2 startup signatures, the retry suppression stays aligned with that new `cpu_unsupported` detail, and the frontend guidance/i18n mapping matches the daemon contract change. The packaging bump and lockfile update also look consistent with the intended Windows baseline runtime fix. Nice follow-through on tightening the matcher and covering the boundary cases in tests.

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

**Comment by lefarcen** [comment]
Re-requesting after 7b7f12621 — a hash-only commit refreshing `nix/pnpm-deps.nix` for the 0.0.25 lockfile (values taken from the nix_validation run's `got:` output). No source changes since the approved head 525ecad25.

**Review by nettee — APPROVED** [review]
@lefarcen I re-reviewed the current head and verified the narrowed `cpu_unsupported` classifier, retry suppression, contract update, and AMR guidance/i18n mapping still line up cleanly. I also checked the final `nix/pnpm-deps.nix` refresh on top of the previously approved source changes, and it matches the lockfile-only package bump for `@powerformer/vela-cli@0.0.25`. Nice job closing the classifier boundary issue and carrying the packaging fix through to the follow-up hash refresh.

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

**Comment by lefarcen** [comment]
Re-requesting review after resolving merge conflicts with main. The conflict was purely the vela-cli version bump racing with main: this branch pinned `@powerformer/vela-cli` 0.0.25 while main moved to 0.0.26 (released 07-17 from vela main, baseline build confirmed in its CI log: `--target=windows-x64-baseline`). Resolution takes main's side for all three conflicted files (`tools/pack/package.json`, `pnpm-lock.yaml`, `nix/pnpm-deps.nix`) — the version bump this PR carried is superseded; the classification fix + UI card are unchanged. Also verified the merge kept `isCpuUnsupportedCrashText` ordered before the fatal_rpc_error close-reason promotion despite #5882 touching the same file, and the focused daemon suites pass (121/121: run-failure-classification, run-retry-policy).

File: apps/daemon/src/run-failure-classification.ts [hunk 3]
```diff
@@ -495,6 +495,21 @@ function isProcessCrashText(text: string): boolean {
...
+// AVX-but-not-AVX2 Sandy/Ivy Bridge cores). Matched only on precise signals:
+// the `no_avx2` CPU-feature line Bun's crash banner always prints on such
+// machines, or Windows STATUS_ILLEGAL_INSTRUCTION as either the hex
+// 0xC000001D or Go/Node's decimal exit-status rendering 3221225501. A bare
+// "Illegal instruction" line is deliberately NOT matched: any unrelated
+// SIGILL (a runtime bug on an AVX2-capable machine) would then be mislabeled
+// as a processor limitation and lose its retry. The same binary on the same
+// CPU fails deterministically, so cpu_unsupported must never be auto-retried.
+function isCpuUnsupportedCrashText(text: string): boolean {
+  return /\bno_avx2\b|0xc000001d|\b3221225501\b/i.test(text);
```
Comment (nettee) [inline apps/daemon/src/run-failure-classification.ts]: `0xC000001D` / `3221225501` on their own are still just generic Windows `STATUS_ILLEGAL_INSTRUCTION`, so promoting every match here to `cpu_unsupported` is broader than the AMR AVX2 startup shape this PR has evidence for. The changed tests below lock that behavior in for a bare `start opencode server: exit status 3221225501` path, which means any other Windows agent crash with the same status will now suppress retry and route the web to the new "update Open Design" guidance even if the real cause is an unrelated SIGILL or a user-installed CLI bug. Please keep the strong `no_avx2` signal, but gate the raw status-code fallback on the bundled AMR/opencode startup context (for example `input.agentId === 'amr'` plus the `start opencode server` text), or leave bare `0xC000001D` / `3221225501` on the existing `fatal_rpc_error` / `process_crashed` path, and add a negative test for a non-AMR or generic illegal-instruction exit.

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

**Review by nettee — CHANGES_REQUESTED** [review]
I found one blocking issue in the current `cpu_unsupported` classifier. The AMR-specific fix looks right, but the raw Windows illegal-instruction fallback is still broader than the startup shape this PR can actually prove.

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

File: apps/daemon/src/run-failure-classification.ts (hunk 3 above)
Comment (lefarcen) [reply apps/daemon/src/run-failure-classification.ts]: Adopted in 83d7ffd78 — verified against the evidence set first: every bannerless production trace carries vela's wrapper text (`start opencode server: opencode exited before readiness: exit status 3221225501`), so gating the raw status-code forms on that context costs zero recall while removing the false-positive surface you describe. `no_avx2` stays unconditional (the CPU-feature line is itself the proof, and it also covers the direct-spawn byok-opencode path where stderr is captured with the banner). I chose the text gate over `input.agentId === 'amr'` because the wrapper string is the thing the 30 traces actually share, and it keeps the helper pure. Added the negative test: `codex acp bridge exited: exit status 3221225501` (no opencode startup context) stays retryable `fatal_rpc_error`. Focused suites pass 122/122.

**Review by AmyShang-alt — APPROVED** [review]
QA validation passed for the PR #5733 AVX2/cpu_unsupported scope on current head `83d7ffd78ee183cb81ee97a264b30d4de70cd3e1`.

Evidence checked:
- Confirmed the classifier now keeps `cpu_unsupported` narrowly scoped: `no_avx2` remains direct evidence, while raw Windows `STATUS_ILLEGAL_INSTRUCTION` (`0xC000001D` / `3221225501`) is only accepted inside the bundled opencode startup/readiness context.
- Confirmed bare or out-of-context illegal-instruction/status-code crashes stay on the existing retryable `fatal_rpc_error` path instead of showing unsupported-processor guidance.
- Ran targeted daemon validation: `pnpm --filter @open-design/daemon exec vitest run tests/run-failure-classification.test.ts tests/run-retry-policy.test.ts --reporter=verbose` -> PASS, 2 files / 122 tests.
- Ran targeted web validation: `pnpm --filter @open-design/web exec vitest run tests/runtime/amr-guidance.test.ts tests/state/config.test.ts --reporter=verbose` -> PASS, 2 files / 83 tests.
- Mac follow-up for the prior Windows-local failures: `run-failure-detail-status.test.ts` and `run-failure-detail-persisted-message.test.ts` passed locally. The remaining `run-resume-on-failure.test.ts` failure on an older local head was a deterministic resume fixture expectation mismatch (`succeeded` vs expected `failed`), not the Windows timeout shape and not part of this AVX2 classifier/regression surface.

QA conclusion: acceptable for the AVX2 crash handling / AMR guidance behavior. CI for current head is green aside from expected skipped jobs, and the previous broad matcher blocker is addressed by the latest narrowing and tests.

**Review by nettee — APPROVED** [review]
